### PR TITLE
5118 Create Javascript hook mechanism

### PR DESCRIPTION
**Pull request by abanink**
hook system for JS to allow addons to register their JS hook
in analogy of php addons, the js addons must be named addon/`addon name`/`addon name`.js

This is a prerequisite to solve #5118 in the for the MathJax addon

### CODE REVIEW

File: view/js/main.js [hunk 1]
```diff
@@ -395,6 +395,19 @@ function NavUpdate() {
 	timer = setTimeout(NavUpdate, updateInterval);
 }
 
+function callAddonHooks(typeOfHook) {
```
Comment (MrPetovan) [inline view/js/main.js]: Can you please move this function declaration to `addon-hooks.js`?

Additionally, can you please add a comment block to explain what this function does?

File: view/js/addon-hooks.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+var addon_hooks={};
+
+function Addon_registerHook( type, hookfnstr )
```
Comment (MrPetovan) [inline view/js/addon-hooks.js]: What is `hookfnstr` supposed to be?

Can you please add a comment block to explain what the function does?

File: view/templates/head.tpl [hunk 3]
```diff
@@ -44,6 +44,10 @@
 <script type="text/javascript" src="view/asset/perfect-scrollbar/js/perfect-scrollbar.jquery.min.js" ></script>
 <script type="text/javascript" src="view/js/acl.js" ></script>
 <script type="text/javascript" src="view/asset/base64/base64.min.js" ></script>
+<script type="text/javascript" src="view/js/addon-hooks.js" ></script>
+{{foreach $addon_hooks as $addon_hook}}
```
Comment (MrPetovan) [inline view/templates/head.tpl]: What if `$addon_hooks` isn't set by any addon?

**Comment by MrPetovan** [comment]
Additionally, can you please elaborate the new JS hook concepts in `doc/Addons.md`?

File: view/templates/head.tpl (hunk 3 above)
Comment (abanink) [reply view/templates/head.tpl]: `addon_hooks` is declared and initialised as an empty array in `addon-hooks.js`, so in that case no additional .js files will be added in `head.tpl`

Comment (MrPetovan) [reply view/templates/head.tpl]: Double curly braces `{{ }}` denotes Smarty tags, and any variable mentioned between them are PHP template variables. This is completely independent from the Javascript variable.

**Pull request merged**